src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -327,19 +350,59 @@ class RemoteMachineTrainingService implements TrainingService {
         return deferred.promise;
     }
 
-    public cleanUp(): Promise<void> {
+    public async cleanUp(): Promise<void> {
         this.log.info('Stopping remote machine training service...');
         this.stopping = true;
+        let timeoutId: NodeJS.Timer;
+        const delay: Promise<{}> = new Promise((resolve: Function, reject: Function): void => {
+            timeoutId = setTimeout(
+                () => { reject(new Error('TrainingService cleanup timeout.')); },
+                10000);
+        });
+        await Promise.race([delay, this.cleanupConnections()]).finally(() => {
+            clearTimeout(timeoutId);
+        });
+    }
+    
+    /**
+     * stop gpu_metric_collector process in remote machine and remove unused scripts
+     */
+    private async cleanupConnections(): Promise<void> {
+        try{
+            for (const [rmMeta, client] of this.machineSSHClientMap.entries()) {
+                let jobpidPath: string = path.join(this.getRemoteScriptsPath(), 'pid');
+                await SSHClientUtility.remoteExeCommand(`pkill -P \`cat ${jobpidPath}\``, client);
+                await SSHClientUtility.remoteExeCommand(`rm -rf ${this.getRemoteScriptsPath()}`, client);
+            }
+        }catch (error) {
+            this.log.error(`Cleanup connection exception, error is ${error.message}`);
+        }
 
         return Promise.resolve();
+    } 
+    
+    private async generateGpuScript(): Promise<string> {
+        let localGpuMetricFolder: string = `/tmp/nni/scripts/${uniqueString(5)}`;
```

Comment:
Don't suggest hard-code gputMetrics folder here. You should use os.tmpdir() to refer temporary folder. 
And also, don't create  common /tmp/nni folder directly, since there's potential permission issue. Create /tmp/{current_user_name}/nni instead, otherwise the owner of /tmp/nni is you (current user of nnictl), but other users on the same machine cannot create /tmp/nni folder because they're not owner.

Reply:
Suggest to use fixed path,  random path is not good for debug, especially on the customer environment that we can not access.

Reply:
use /tmp/{USER}/nni as the temp path.